lib/Common/Memory/PageAllocator.cpp

```diff
@@ -561,23 +571,47 @@ PageSegmentBase<T>::DecommitFreePages(size_t pageToDecommit)
     Assert(pageToDecommit != 0);
     char * currentAddress = this->address;
 
-    uint decommitCount = 0;
+    uint decommitCount = 0, lastDecommitCount = 0;
+    bool freeBlock = false;
     for (uint i = 0; i < this->GetAvailablePageCount(); i++)
     {
         if (this->TestInFreePagesBitVector(i))
```

Comment:
Considering the new implementation there, could you elaborate on gain and approach?